### ISSUE TITLE
djb2 rule fails with Revil sample

### ISSUE THREAD

**Issue by m4now4r** [mandiant/capa-rules#434]
## Summary

Revil ransomware uses the **djb2** hash algorithm, but the djb2 rule does not detect it. May be the variant because the compiler generates different code.

## Examples

- Sample hash: 9B11711EFED24B3C6723521A7D7EB4A52E4914DB7420E278AA36E727459D59DD (found in the Kaseya incident)
- File offset: `0x6555`
- Disassembly code:
![image](https://user-images.githubusercontent.com/27551832/126414890-f9e749d6-05d9-4a9b-9e75-c09560a614f6.png)
- Pseudo-code:
![image](https://user-images.githubusercontent.com/27551832/126415006-b29d41f9-d58d-4e86-b4e9-d111fca27d04.png)

## Possible improvements

I tried to clone a new rule and replace` shl` with:
```
- description: hash * 33
- mnemonic: imul
- number: 33
```

Thanks team for great tool!

**Comment by williballenthin** [comment]
@cdong1012 want to take this?

**Comment by mr-tz** [comment]
Good suggestion, adding a `basic block` statement per your suggestion should better capture this.

**Comment by mr-tz** [comment]
This was added in #656! Thanks to @Still34!

```yml
hash data using djb2 (2 matches)
namespace   data-manipulation/hashing/djb2
author      awillia2@cisco.com, still@teamt5.org
scope       function
mbc         Data::Non-Cryptographic Hash [C0030]
references  https://twitter.com/r3c0nst/status/1392405576131436546, http://www.cse.yorku.ca/~oz/hash.html
function @ 0x2C96555
  and:
    instruction:
      and:
        mnemonic: mov @ 0x2C9655B
        number: 0x1505 @ 0x2C9655B
    or:
      instruction:
        and:
          number: 0x21 @ 0x2C96562
          or:
            mnemonic: imul @ 0x2C96562
function @ 0x2C96573
  and:
    instruction:
      and:
        mnemonic: mov @ 0x2C96579
        number: 0x1505 @ 0x2C96579
    or:
      instruction:
        and:
          number: 0x21 @ 0x2C96580
          or:
            mnemonic: imul @ 0x2C96580
```
